src/wing_segmenter/cli.py

```diff
@@ -0,0 +1,125 @@
+import argparse
+
+def main():
+    parser = argparse.ArgumentParser(
+        prog='wingseg',
+        description="Wing Segmenter CLI",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    subparsers = parser.add_subparsers(title='Commands', dest='command', required=True)
+
+    # Subcommand: segment
+    segment_parser = subparsers.add_parser(
+        'segment',
+        help='Segment images and store segmentation masks.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    # Required argument
+    segment_parser.add_argument('--dataset', required=True, help='Path to dataset images')
+
+    # Resizing options
+    resize_group = segment_parser.add_argument_group('Resizing Options')
+
+    # Dimension specifications
+    resize_group.add_argument('--size', nargs='+', type=int,
+                              help='Target size. Provide one value for square dimensions or two for width and height.')
+
+    # Resizing mode
+    resize_group.add_argument('--resize-mode', choices=['distort', 'pad'], default=None,
+                              help='Resizing mode. "distort" resizes without preserving aspect ratio, "pad" preserves aspect ratio and adds padding if necessary.')
+
+    # Padding options (to preserve aspect ratio)
+    resize_group.add_argument('--padding-color', choices=['black', 'white'], default=None,
+                              help='Padding color to use when --resize-mode is "pad".')
+
+    # Interpolation options
+    resize_group.add_argument('--interpolation', choices=['nearest', 'linear', 'cubic', 'area', 'lanczos4', 'linear_exact', 'nearest_exact'],
+                              default='area',
+                              help='Interpolation method to use when resizing. For upscaling, "lanczos4" is recommended.')
+
+    # Bounding box padding option
+    bbox_group = segment_parser.add_argument_group('Bounding Box Options')
+    bbox_group.add_argument('--bbox-padding', type=int, default=None,
+                            help='Padding to add to bounding boxes in pixels. Defaults to no padding.')
+
+
+    # Output options within mutually exclusive group
+    output_group = segment_parser.add_mutually_exclusive_group()
+    output_group.add_argument('--outputs-base-dir', default=None, help='Base path to store outputs.')
+    output_group.add_argument('--custom-output-dir', default=None, help='Fully custom directory to store all output files.')
```

Comment:
What exactly is the difference between these?

Reply:
The first one `--outputs-base-dir` will create a folder for outputs within the specified directory with a unique identifier based on the args passed and dataset processed. Makes it useful for testing parameters like resize and bbox padding and enables the `scan-runs` table to give an overview of the outputs within `--outputs-base-dir`.

The second one `--custom-output-dir` will put outputs directly in the specified directory for more user control.

Reply:
This explanation should probably be added somewhere, so someone else won't have the exact same question.